src/ds/address.h

```diff
@@ -81,7 +71,10 @@ namespace snmalloc
 #if __has_builtin(__builtin_align_down)
       return static_cast<T*>(__builtin_align_down(p, alignment));
 #else
-      return pointer_cast<T>(bits::align_down(address_cast(p), alignment));
+      static_assert(
+        aal_supports<IntegerPointers>,
```

Comment:
I am not convinced by this. If into intptr_t exposes an address, not an offset, we can handle this case for CHERI too.

Reply:
The intent was that `address_t` would become `vaddr_t`, and not `intptr_t` on CHERI, so that it truly was an address.

Reply:
Does `reinterpret_cast<T*>(bits::align_down(reinterpret_cast<uintptr_t>(p), alignment));` work?  If so, the comment is misleading: we can realign via `bits` without integer pointers, we can't realign via bits if we explicitly cast to an address first.

Reply:
That's fair; I'll drop the assert and change the casts.